File: Data Structures & Algorithms/duplicate-integer/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let val = {};

        // iterate through the entire array
        // if its a new value then add to val hashmap and give it value 1
        // otherwise increment the current value
        for(let i = 0; i < nums.length; i++){
            
            
            if(val[nums[i]] === undefined){
                val[nums[i]] = 1;
            }else{
                val[nums[i]] += 1;
            }
        }

        // check if the object has a dup
        for(const [key, count] of Object.entries(val)){
            if(count > 1){
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-5.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        counter = 0
        maxCounter = 0
        for i in range(len(nums)):
            # check if currVal is a 1
            currVal = nums[i]
            if currVal == 1:
                # increment it
                counter += 1
                if counter > maxCounter:
                    maxCounter = counter
            else:
                # if not a 1
                counter = 0

            

        return maxCounter